src/vehicle.cpp

```diff
@@ -874,6 +874,26 @@ bool vehicle::has_engine_type_not( const itype_id &ft, bool const enabled ) cons
     return false;
 }
 
+bool vehicle::has_engine_conflict( const vpart_info *possible_conflict,
+                                   std::string &conflict_type ) const
+{
+    std::vector<std::string> new_excludes = possible_conflict->engine_excludes();
+    bool has_conflict = false;
+
+    for( size_t e = 0; e < engines.size(); ++e ) {
```

Comment:
Before iterating through all engines, test if `new_excludes` is empty. If it is empty there is no possibility of conflict and iterating is pointless.

Reply:
Thanks, good catch.

Reply:
Fixed.